src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java

```diff
@@ -197,21 +199,56 @@ private void extractCustomFields(PaymentWorksVendor stgNewVendor, PaymentWorksNe
+                }
+            }
+        } else if (paymentWorksFormModeService.shouldUseForeignFormProcessingMode()) {
+            processEachCustomFIeldsTheNewWay(stgNewVendor, pmwNewVendorDetailDTO, customFieldErrors);
+        }
+    }
+
+    protected void processEachCustomFIeldsTheNewWay(PaymentWorksVendor stgNewVendor, PaymentWorksNewVendorRequestDetailDTO pmwNewVendorDetailDTO,
+            List<String> customFieldErrors) {
+        for (PaymentWorksCustomFieldDTO customField : pmwNewVendorDetailDTO.getCustom_fields().getCustom_fields()) {
+            PaymentWorksFieldMapping fieldMapping = findPaymentWorksFieldMapping(customField.getField_id());
             if (ObjectUtils.isNotNull(fieldMapping)) {
-                setPaymentWorksVendorCustomFieldAttributeToReceivedValue(stgNewVendor, paymentWorksCustomFieldId, fieldMapping, customFieldsDTOMap, customFieldErrors);
-            } 
-            else {
+                Map<String, String> customFieldsDTOMap = new HashMap();
+                customFieldsDTOMap.put(customField.getField_id(), getValueOutOfPaymentWorksCustomFieldDTO(customField, fieldMapping));
+                setPaymentWorksVendorCustomFieldAttributeToReceivedValue(stgNewVendor, customField.getField_id(), fieldMapping, 
+                        customFieldsDTOMap, customFieldErrors);
+            } else {
                 stgNewVendor.setCustomFieldConversionErrors(true);
-                customFieldErrors.add(MessageFormat.format(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.NEW_VENDOR_REQUEST_CUSTOM_FIELD_MISSING_ERROR_MESSAGE), paymentWorksCustomFieldId));
-                LOG.error("processEachPaymentWorksCustomFieldReceived: Unable to find KFS staging table column for PaymentWorks custom field '" + paymentWorksCustomFieldId + "'");
+                customFieldErrors.add(MessageFormat.format(getConfigurationService().getPropertyValueAsString(
+                        PaymentWorksKeyConstants.NEW_VENDOR_REQUEST_CUSTOM_FIELD_MISSING_ERROR_MESSAGE), customField.getField_id()));
+                LOG.error("processEachCustomFIeldsTheNewWay: Unable to find KFS staging table column for PaymentWorks custom field '" + 
+                        customField.getField_id() + "'");
             }
         }
     }
     
+    protected String getValueOutOfPaymentWorksCustomFieldDTO(PaymentWorksCustomFieldDTO dto, PaymentWorksFieldMapping fieldMapping) {
+        if (StringUtils.equalsIgnoreCase(fieldMapping.getCustomAttributeValueToUse(), PaymentWorksConstants.CustomAttributeValueToUse.FILE)) {
+            return dto.getFile();
+        } else if (StringUtils.equalsIgnoreCase(fieldMapping.getCustomAttributeValueToUse(), PaymentWorksConstants.CustomAttributeValueToUse.FIELD_VALUE)) {
+            return dto.getField_value();
+        } else {
+            throw new IllegalArgumentException("The custom attrbute value to use is not value: " + fieldMapping.getCustomAttributeValueToUse());
```

Comment:
There is a misspelling in the exception message. I am also not sure the message makes sense. Should the "not value" at the end be removed?

Reply:
that should be not valid, I fixed